TheMovieDB/TheMovieDB/MVP/Views/HomeScreenVC.swift

```diff
@@ -0,0 +1,140 @@
+        collectionView.delegate = self
+        view.addSubview(collectionView)
+        collectionView.frame = view.bounds
+        
+        searchBar.searchTextField.textColor = UIColor.green
+        searchBar.searchTextField.backgroundColor = UIColor.black
+        searchBar.searchTextField.leftView?.tintColor = UIColor.green
+        searchBar.subviews[0].tintColor = UIColor.green
+        
+        switch deviceIdiom {
+        case .pad:
+            tableView.isHidden = true
+            collectionView.isHidden = false
+        default:
+            tableView.isHidden = false
+            collectionView.isHidden = true
+        }
+        
+        networkProvider.getMovies(completion: { [weak self] result in self?.movies = result
+            
+            DispatchQueue.main.async { [weak self] in self?.tableView.reloadData()
+            }
+        })
+    }
+}
+
+extension HomeScreenVC: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return movies.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let movie = movies[indexPath.row]
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CollectionCell.identifier, for: indexPath) as? CollectionCell else {
+            return UICollectionViewCell()
+        }
+        
+        collectionCell.setMovieCell(movie: movie, cell: cell)
+        return cell
+    }
+}
+
+
+extension HomeScreenVC: UICollectionViewDelegate {
+    
+}
+
+/*
+ 0) LISTO: crear clase collectionviewcell
+ 1) crear collectionView, ver márgenes, diseñarla un poco,
+ 2) LISTO: ocultar tableview y mostrar collection view según device, en el viewdidload
+ 3) LISTO: setear los valores que van en la collectionview según las celdas
+ */
+
+
+
+extension HomeScreenVC: UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return movies.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let movie = movies[indexPath.row]
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "MovieCell", for: indexPath) as? TableCell
+        else {
+            return UITableViewCell()
+        }
+        
+        tableCell.setMovieCell(movie: movie, cell: cell)
+        
+        return cell
+    }
+}
+
+extension HomeScreenVC: UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        selectedMovie = movies[indexPath.row]
+        performSegue(withIdentifier: "goToMovieDetailVC", sender: selectedMovie)
+    }
+}
+
+extension HomeScreenVC: UISearchBarDelegate {
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        movies = networkProvider.movies
```

Comment:
You need to re-think this implementation because you are assigning two time to the same variable